app/models/node.rb

```diff
@@ -18,26 +20,42 @@ class Node < ApplicationRecord
   # we don't want to find just the UserTags associated with Authz; that's not v useful on its own.
   # has_many :anchored_authz_tags, through: :anchoring_authz_tags, source: :target, source_type: "UserTag"
 
+  has_many :system_tag_decls, as: :anchored
+  has_many :system_user_tags, through: :system_tag_decls, source: :target, source_type: "UserTag"
+
   has_many :node_authz_reads, foreign_key: :base_node_id
 
   after_create :set_tags
 
   # scope :is_top_post, -> (x) { where(is_top_post: x) }
   # scope :genesis, -> (x) { where(genesis_id: x.id) }
 
-  def children(user_id)
-    user_id_comp = user_id.nil? ? "IS" : "="
-    # return Node.where("id in (
-    #   SELECT nwc.id FROM nodes_user_sees nus
-    #   JOIN node_with_children nwc ON nwc.id = nus.base_node_id
-    #   WHERE nwc.base_node_id = ? AND rel_depth = 1 AND nus.user_id #{user_id_comp} ?
-    # )", self.id, user_id)
-    # alt implementation that might/should be faster
+  def children(user, limit: 1000)
+    nar = Arel::Table.new :nar
+    Node.find_by_sql(Node.nodes_readable_by(user).where(nar[:parent_id].eq(id)).take(limit))
+  end
+
+  def children_elegaint(user)
+    user_id_comp = user.nil? ? "IS" : "="
+    Node.where("id in (
+      SELECT nwc.id FROM nodes_user_sees nus
+      JOIN node_with_children nwc ON nwc.id = nus.base_node_id
+      WHERE nwc.base_node_id = ? AND rel_depth = 1 AND nus.user_id #{user_id_comp} ?
+    )", self.id, user.id)
+  end
+
+  def children_elegaint_parent(user)
```

Comment:
note: most of these will get cut. they're still here to support benchmark.rb so we can gather data and stuff